src/main/java/com/svydovets/bibirnate/cache/impl/CacheImpl.java

```diff
@@ -0,0 +1,172 @@
+package com.svydovets.bibirnate.cache.impl;
+
+
+import com.svydovets.bibirnate.cache.Cache;
+import com.svydovets.bibirnate.exception.EntityKeyNotPresentException;
+import com.svydovets.bibirnate.exception.EntityNotPresentException;
+import com.svydovets.bibirnate.exception.EntityTypeNotPresentException;
+import com.svydovets.bibirnate.exception.IdNotPresentException;
+import com.svydovets.bibirnate.util.EntityKey;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The {@link CacheImpl} implements {@link Cache} and provides methods for working with cache.
+ */
+public class CacheImpl implements Cache {
+
+    private static final String CACHE_PUT = "Cache#put:";
+    private static final String CACHE_GET = "Cache#get:";
+
+    private final Map<EntityKey<?>, Object> cacheMap;
+    private boolean enabled;
+
+    /**
+     * Creates {@link CacheImpl} with enabled access to {@link CacheImpl#cacheMap}.
+     */
+    public CacheImpl() {
+        cacheMap = new HashMap<>();
+        this.enabled = true;
+    }
+
+    /**
+     * Creates {@link CacheImpl} with enabled or disabled access to {@link CacheImpl#cacheMap}.
+     * {@link  CacheImpl#enabled} adjust it.
```

Comment:
Sorry, but I don't understand what `{@link  CacheImpl#enabled} adjust it.` should mean.